### DISCUSSION TITLE
sysfs write failed rbd: map failed: (95) Operation not supported

### DISCUSSION THREAD

**Discussion by jerryjedi** [rook/rook#10642] [Q&A] [answered]
I installed rook ceph 1.9.8 on a new kubernetes environment. 
I checked ceph status using ceph tool, seems all are correct. 

But when i deployed sample application(wordpress from ceph.io) to consume the storage, I met the problem.
The log of pod wordpress-mysql-xxxx:
MountVolume.MountDevice failed for volume "pvc-a87c8e97-ad68-4772-8eb5-b22c404461a2" : rpc error: code = Internal desc = rbd: map failed with error an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.233.15.160:6789,10.233.6.11:6789,10.233.39.91:6789 --keyfile=***stripped*** map replicapool/csi-vol-a6402cb9-0d69-11ed-a315-d64cd13fa192 --device-type krbd --options noudev], rbd error output: rbd: sysfs write failed rbd: map failed: (95) Operation not supported

The OS is ubuntu, and kernal is 4.17.19, all installation process are followed the documents from rook.io. 

I read the issue: #2397, I don't think it's the same problem. 

Welcome any suggestion and thanks.

**Comment by Madhu-1** [comment, marked as answer]
@jerryjedi can you please check dmesg logs on the node where the map is failing?

**Reply by jerryjedi** [reply]
I found some red line in the output of "dmesg":
[   23.737300] IPVS: rr: TCP 10.233.0.3:53 - no destination available
[   23.737318] IPVS: rr: TCP 10.233.0.3:53 - no destination available
[   23.750956] IPVS: rr: TCP 10.233.0.3:53 - no destination available

I dump the output in the  file. 

[dmesg_msg.txt](https://github.com/rook/rook/files/9197302/dmesg_msg.txt)

I'm wonder is this the error message related to my problem?

**Reply by Madhu-1** [reply]
Is it from the same node where the map is failing? If yes, can you please exec into the rbdplugin on the node and run the rbd map manually? and also run `rbd info replicapool/csi-vol-a6402cb9-0d69-11ed-a315-d64cd13fa192` and paste the output

```
rbd --id csi-rbd-node -m 10.233.15.160:6789,10.233.6.11:6789,10.233.39.91:6789 --key=xxxx map replicapool/csi-vol-a6402cb9-0d69-11ed-a315-d64cd13fa192 --device-type krbd --options noudev
```

You can get the key value from the toolbox pod.

```
ceph auth get-key client.csi-rbd-node
```

**Reply by jerryjedi** [reply]
<img width="963" alt="1" src="https://user-images.githubusercontent.com/7875026/181403685-279b9011-5e51-4462-858d-c03584b55c9a.png">
I run the command , got the error message:
**rbd: failed to set udev buffer size: (1) Operation not permitted
rbd: sysfs write failed
In some cases useful info is found in syslog - try "dmesg | tail".
rbd: map failed: (30) Read-only file system**

The detail info is in the attached file.

**Reply by Madhu-1** [reply]
i dont have anything on the attached file https://github.com/rook/rook/discussions/10642#discussioncomment-3259525, give you give a try with https://rook.io/docs/rook/latest/Troubleshooting/direct-tools/#start-the-direct-mount-pod? and also can you try with different version of ubuntu looks to be  ENV issue.

**Reply by jerryjedi** [reply]
I upgraded the os kernel from 4.17.19 to 4.20.17, then the problem seemed to vanish into thin air.But I don't know the root cause.

**Comment by zhucan** [comment]
@jerryjedi The ceph client in the csi driver match to the  ceph that deployed by rook?

**Comment by ravikakadia** [comment]
@Madhu-1 I am facing the same issue. Here is my dmesg output from the node where wordpress pod is scheduled.
[dmesg.txt](https://github.com/user-attachments/files/17704291/dmesg.txt)
I do not see any errors in the dmesg file.

Also, rbd pod not able to connect to the cluster
```
[root@worker1954467 /]# rbd --id csi-rbd-node -m 10.100.245.171:6789,10.104.205.146:6789,10.96.21.73:6789 --keyfile=AQAO4S1nBs7FKhAAInoa6o6ENGVzfRMEviNypg== map replicapool/csi-vol-a7df2433-6926-4078-b8a6-d3a933726999 --device-type krbd --options noudev
rbd: couldn't connect to the cluster!
```
```
[root@worker1954467 /]# rbd info replicapool/csi-vol-a7df2433-6926-4078-b8a6-d3a933726999
unable to get monitor info from DNS SRV with service name: ceph-mon
rbd: couldn't connect to the cluster!
2024-11-11T15:36:18.433+0000 7fca9aca3d80 -1 failed for service _ceph-mon._tcp
2024-11-11T15:36:18.433+0000 7fca9aca3d80 -1 monclient: get_monmap_and_config cannot identify monitors to contact

```

**Reply by Madhu-1** [reply]
Please pass the monitor , id and keyring to the info command as well and check if you have connectivity with ceph

**Reply by ravikakadia** [reply]
Also when I use direct mount pod and try to map rbd, I encounter the same issue @Madhu-1 
```
[root@worker1954467 /]# rbd map replicapool/test
rbd: sysfs write failed
In some cases useful info is found in syslog - try "dmesg | tail".
rbd: map failed: (95) Operation not supported
[root@worker1954467 /]#

```

**Reply by Madhu-1** [reply]
@ravikakadia what is the kernel version you are using? you might need to run with rbd debug logs to see what is happening.

**Reply by ravikakadia** [reply]
```
[admin@controller195442: ~] $ uname -a
Linux controller195442 4.15.0-1011-fips #12-Ubuntu SMP Tue Jul 2 21:04:45 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

```

**Reply by ravikakadia** [reply]
[rbd_debug_logs.txt](https://github.com/user-attachments/files/17905633/rbd_debug_logs.txt)
PFA rbd debug logs. @Madhu-1

**Reply by Madhu-1** [reply]
@ravikakadia is it possible to update and try with new kernel version?

**Reply by ravikakadia** [reply]
Ok, I will try out. We have FIPS enabled OS, so hope that is not causing any issues.

**Reply by ravikakadia** [reply]
@Madhu-1 I have upgraded my ubuntu to 20.04 and issue is resolved. One more difference I have is , OS upgrade has disabled FIPS also. I will try enabling the FIPS again and verify rbd. Thanks for your help and pointers.

**Comment by ravikakadia** [comment]
@Madhu-1  I was providing key value for keyfile hence I got this error  - unable to connect to the cluster. After fixing above commands I see output for info command and same error -sysfs write failed .
```

[root@controller195442 /]# rbd info --id csi-rbd-node -m 10.96.155.83:6789,10.98.85.205:6789,10.96.57.159:6789 --key=AQAbMzxn233KFhAAwce+ux/hDMg7vSAF6gLvjA== replicapool/csi-vol-5310572b-0366-4589-bd8c-31d376121705
rbd image 'csi-vol-5310572b-0366-4589-bd8c-31d376121705':
        size 20 GiB in 5120 objects
        order 22 (4 MiB objects)
        snapshot_count: 0
        id: 6e202b04c6d5
        block_name_prefix: rbd_data.6e202b04c6d5
        format: 2
        features: layering
        op_features:
        flags:
        create_timestamp: Tue Nov 19 10:31:26 2024
        access_timestamp: Tue Nov 19 10:31:26 2024
        modify_timestamp: Tue Nov 19 10:31:26 2024
[root@controller195442 /]# rbd --id csi-rbd-node -m 10.96.155.83:6789,10.98.85.205:6789,10.96.57.159:6789 --key=AQAbMzxn233KFhAAwce+ux/hDMg7vSAF6gLvjA== map replicapool/csi-vol-5310572b-0366-4589-bd8c-31d376121705 --device-type krbd --options noudev
rbd: sysfs write failed
In some cases useful info is found in syslog - try "dmesg | tail".
rbd: map failed: (95) Operation not supported
[root@controller195442 /]# dmesg | tail
[322068.034297] IPv6: ADDRCONF(NETDEV_UP): lxcef6a072856bf: link is not ready
[322068.061111] eth0: renamed from tmpbb271
[322068.087763] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[322068.087780] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[322068.087841] IPv6: ADDRCONF(NETDEV_CHANGE): lxcef6a072856bf: link becomes ready
[322103.301330] IPv6: ADDRCONF(NETDEV_UP): lxcefd063ab1cb7: link is not ready
[322103.336581] eth0: renamed from tmp20517
[322103.364402] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[322103.364425] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[322103.364498] IPv6: ADDRCONF(NETDEV_CHANGE): lxcefd063ab1cb7: link becomes ready
[root@controller195442 /]#

```
